spec/lib/content_exporter_spec.rb

```diff
@@ -0,0 +1,199 @@
+      title: "I am the title for Group two subgroup one",
+      items: [{ id: "0006", text: "This is a row that only has text, it will appaer like a paragraph" }],
+    },
+  },
+}
+
+locale_form_groups_fixture = {
+  group_one: { title: "I am the title for Group one" },
+  group_two: { title: "I am the title for Group two" },
+}
+
+RSpec.describe "ContentExporter" do
+  describe "#extract_results_links" do
+    let(:results_rows) { ContentExporter.extract_results_links }
+
+    results_row_fixture = [
+      { id: "0001",
+        group_title: "I am the title for Group one",
+        href: "",
+        show_to_nations: "",
+        subgroup_title: "I am the title for Group one subgroup one",
+        support_and_advice: false,
+        text: "This is a row that only has text, it will appaer like a paragraph" },
+      { id: "0002",
+        group_title: "I am the title for Group one",
+        href: "http://test.stubbed.gov.uk",
+        show_to_nations: "",
+        subgroup_title: "I am the title for Group one subgroup one",
+        support_and_advice: false,
+        text: "This is a row has text and an href, it will appaer as an anchor tag" },
+      { id: "0003",
+        group_title: "I am the title for Group one",
+        href: "http://test.stubbed.llyw.cymru",
+        show_to_nations: "Wales",
+        subgroup_title: "I am the title for Group one subgroup one",
+        support_and_advice: false,
+        text: "This is a row has text, an href and group criteira, it will appear as an anchor tag if the user's answers match the criteria" },
+      { id: "0004",
+        group_title: "I am the title for Group one",
+        href: "http://test.stubbed.gb.gov.uk",
+        show_to_nations: "Wales OR Scotland OR England",
+        subgroup_title: "I am the title for Group one subgroup one",
+        support_and_advice: true,
+        text: "This is a row has text, an href and multiple group criteira, it will appear as an anchor tag if the user's answers match the more complex criteria" },
+      { id: "0005",
+        group_title: "I am the title for Group one",
+        href: "",
+        show_to_nations: "",
+        subgroup_title: "I am the title for Group one subgroup two",
+        support_and_advice: false,
+        text: "This is a row that only has text, it will appaer like a paragraph" },
+      { id: "0006",
+        group_title: "I am the title for Group two",
+        href: "",
+        show_to_nations: "",
+        subgroup_title: "I am the title for Group two subgroup one",
+        support_and_advice: false,
+        text: "This is a row that only has text, it will appaer like a paragraph" },
+    ]
+
+    before do
+      allow(I18n).to receive(:t).with("results_link") { locale_results_link_fixture }
+      allow(I18n).to receive(:t).with("coronavirus_form.groups") { locale_form_groups_fixture }
+    end
+
+    it "returns an array of result link rows" do
+      expect(results_rows).to eq(results_row_fixture)
```

Comment:
Is `match_array` a better fit here, or is ordering important?

Reply:
Ah that cool, tbh i hadn't considered `match_array` good shout.
Humm i think ordering is significant though, we render things in the order they appear in the yaml, and i think content designers will want the sheet to be the "order rendered" too.

So i think we should check for ordered equality here.